components/cli/internal/test/bal_executor.go

```diff
@@ -47,15 +52,54 @@ func SetBalVersion(version string) func(*MockBalExecutor) {
 		balExecutor.version = version
 	}
 }
+func SetBalCurrentDir(currentDir string) func(*MockBalExecutor) {
+	return func(balExecutor *MockBalExecutor) {
+		balExecutor.currentDir = currentDir
+	}
+}
+
+func SetYamlName(name string) func(*MockBalExecutor) {
+	return func(balExecutor *MockBalExecutor) {
+		balExecutor.yamlName = name
+	}
+}
+
+func SetYamlContent(content []byte) func(*MockBalExecutor) {
+	return func(balExecutor *MockBalExecutor) {
+		balExecutor.yamlContent = content
+	}
+}
+
+func SetMetadataJsonContent(content []byte) func(*MockBalExecutor) {
+	return func(balExecutor *MockBalExecutor) {
+		balExecutor.metadataJsonContent = content
```

Comment:
Why do we need json content in bal executor?

Reply:
This is the mock ballerina executor. For unit tests, we need to mock the generation of artifacts files (metadata.json & yaml) to test "Cellery Run" command.

Reply:
Ack, thanks.